ql/src/java/org/apache/hadoop/hive/ql/exec/DDLTask.java

```diff
@@ -5225,7 +5226,8 @@ private int updateFirstIncPendingFlag(Hive hive, ReplSetFirstIncLoadFlagDesc des
       for (String dbName : Utils.matchesDb(hive, dbNameOrPattern)) {
         Database database = hive.getMSC().getDatabase(dbName);
         parameters = database.getParameters();
-        if (ReplUtils.isFirstIncPending(parameters)) {
+        String incPendPara = parameters != null ? parameters.get(ReplUtils.REPL_FIRST_INC_PENDING_FLAG) : null;
+        if (incPendPara != null && (!flag.equalsIgnoreCase(incPendPara))) {
```

Comment:
Same as above.

Reply:
changed to ReplRemoveFirstIncLoadPendFlagDesc ..now just remove if the property is set